### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by stevenhillcox**
The main poll controller is not responsible for fetching the poll, instead of doing it on each sub controller so we need to use a watch on the sub controllers.
(Although angular has implicit watches on scope variable, this doesn't propagate to elements bound to sub properties)

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Views/Routes/Manage.cshtml [hunk 1]
```diff
@@ -60,6 +60,8 @@
                 <br />
                 <span ng-if="poll.NamedVoting">Named Voters</span>
                 <span ng-if="!poll.NamedVoting">Anonymous Voters</span>
+                <br />
+                <span><span ng-if="!poll.OptionAdding">No </span>Voter Option Adding</span>
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Views/Routes/Manage.cshtml]: Can we word this a bit better? E.g. "Voters can/cannot add options"

File: VotingApplication/VotingApplication.Web/Scripts/Controllers/BasicVoteController.js [hunk 2]
```diff
@@ -29,11 +29,12 @@
 
         function activate() {
             $scope.$watch('poll', function () {
-                $scope.options = $scope.poll ? $scope.poll.Options : [];
+                if ($scope.poll) {
+                    $scope.options = $scope.poll.Options;
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/BasicVoteController.js]: A lot of these are doing the same thing. We should pull them up to the VotingPageController in the PollService.getPoll call, and have an extra call for configuring with the poll object, similar to how getVotes.

I think the only one that needs to do additional config is the points vote setting all option.voteValue = 0

File: VotingApplication/VotingApplication.Web/Scripts/Controllers/PointsVoteController.js [hunk 3]
```diff
@@ -33,20 +33,19 @@
...
-                    d.voteValue = 0;
-                });
+                    $scope.options.forEach(function (d) {
+                        d.voteValue = 0;
+                    });
 
-                $scope.totalPointsAvailable = $scope.poll ? $scope.poll.MaxPoints : 0;
-                $scope.maxPointsPerOption = $scope.poll ? $scope.poll.MaxPerVote : 0;
+                    $scope.totalPointsAvailable = $scope.poll ? $scope.poll.MaxPoints : 0;
+                    $scope.maxPointsPerOption = $scope.poll ? $scope.poll.MaxPerVote : 0;
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/PointsVoteController.js]: Now that we are inside the "if ($scope.poll), we don't need to do this check

**Comment by tpkelly** [comment]
:+1:

**Comment by JDawes-ScottLogic** [comment]
:+1: :shipit:

**Pull request merged**